### PR TITLE
Fixes javascript firing. Thanks Andrew.

### PR DESCRIPTION
**Pull request by StephHippo**
Andrew became a Javascript Ninja for like 60 seconds and spotted the document ready issue.

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def postrequisites
   end
 
   def schedulable?
-    course_instances.any?(&:schedulable?)
+    course_instances.any?{|course_instance| course_instance.try(:schedulable?)}
```
Comment (ajm188) [inline app/models/course.rb]: @StephHippo fix your style. Keep hound happy :dog:

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -37,7 +37,8 @@ def set_current_user
 
   # Used for the searching done by both the SchedulerController and CourseInstancesController
   def set_search_date
-    search_date_string = Semester::SAFE_SEARCH_DATES[params[:semester][:semester]][params[:semester][:half]] + " #{params[:semester][:year]}"
+    search_date_string = Semester::SAFE_SEARCH_DATES[params[:semester][:semester]][params[:semester][:half]] +
```
Comment (StephHippo) [inline app/controllers/application_controller.rb]: Ass good as it's gonna get.

File: lib/sis_scraper.rb [hunk 3]
```diff
@@ -3,18 +3,16 @@
...
+  LEN_DEPT_NAME = 4	# Standard length of a department abbreviation
+
   def initialize(download_dir)
-    @download_dir=download_dir
+    @download_dir = download_dir
     @browser = Watir::Browser.new :firefox
   end
 
-  def download_all_sis_data(user,pword)
+  def download_all_sis_data
```
Comment (StephHippo) [inline lib/sis_scraper.rb]: Not sure how this is calculated. Ignoring for now.

**Comment by StephHippo** [comment]
So this def should have been in a separate branch. I apologize.

**Comment by ajm188** [comment]
:+1: merge it. great commit messages.

File: spec/models/course_spec.rb [hunk 4]
```diff
@@ -3,15 +3,16 @@
 RSpec.describe Course, type: :model do
   before do
     @course = FactoryGirl.build(:course, department: "EECS", course_number: 132)
-    @course.course_instances = [FactoryGirl.build(:course_instance, :end_date => Date.today-1), FactoryGirl.build(:course_instance, :end_date=>Date.today+365)]
-    #Testing unhappy paths
+    @course.course_instances = [FactoryGirl.build(:course_instance, :end_date => Date.today - 1),
+                                FactoryGirl.build(:course_instance, :end_date => Date.today + 365)]
```
Comment (ajm188) [inline spec/models/course_spec.rb]: @StephHippo hahahaha

Comment (StephHippo) [inline spec/models/course_spec.rb]: I REFUSE. Rockets 4 lyfe.

**Pull request merged**